overlord/configstate/configcore/services.go

```diff
@@ -231,6 +236,85 @@ func handleServiceDisableConfiguration(_ sysconfig.Device, tr config.ConfGetter,
 			}
 		}
 	}
+	// configure ssh ports
+	if err := handleNetworkConfigSSHPort(dev, tr, opts); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateServiceConfiguration(tr config.ConfGetter) error {
+	// validate the ssh port setting
+	output, err := coreCfg(tr, sshPortOpt)
+	if err != nil {
+		return err
+	}
+	if output == "" {
+		return nil
+	}
+	port, err := strconv.Atoi(output)
+	if err != nil {
+		return err
+	}
+	if port > 65535 || port < 1 {
+		return fmt.Errorf("cannot use port %v: must be in the range 1-65535", port)
+	}
+	return nil
+}
+
+func handleNetworkConfigSSHPort(dev sysconfig.Device, tr config.ConfGetter, opts *fsOnlyContext) error {
+	// see if anything needs to happen
+	var pristineSSHPort, newSSHPort interface{}
+
+	if err := tr.GetPristine("core", sshPortOpt, &pristineSSHPort); err != nil && !config.IsNoOption(err) {
+		return err
+	}
+	if err := tr.Get("core", sshPortOpt, &newSSHPort); err != nil && !config.IsNoOption(err) {
+		return err
+	}
+	if pristineSSHPort == newSSHPort {
+		return nil
+	}
+
+	// ssh.port config has changed, write new config
+	root := dirs.GlobalRootDir
+	if opts != nil {
+		root = opts.RootDir
+	}
+
+	// Note: Only UC20+ supports ussing the "sshd_config.d"
+	// dir. Supporting older systems would be hard because we
+	// would have to merge somehow the UC16 and UC18 config in a
+	// fsOnlyContext
+	if !dev.HasModeenv() {
+		return fmt.Errorf("cannot set ssh port configuration on systems older than UC20")
+	}
+	portConfPath := filepath.Join(root, "/etc/ssh/sshd_config.d/port.conf")
+	if err := os.MkdirAll(filepath.Join(root, "/etc/ssh/sshd_config.d"), 0755); err != nil {
+		return err
+	}
+	if newSSHPort == "" {
+		if err := os.Remove(portConfPath); err != nil && !os.IsNotExist(err) {
+			return err
+		}
+	} else {
+		configStr := fmt.Sprintf("Port %v", newSSHPort)
+		if err := ioutil.WriteFile(portConfPath, []byte(configStr), 0600); err != nil {
+			return err
+		}
+	}
```

Comment:
The EnsureFileState machinery could fit well here.

Reply:
Changed this now